README.md

```diff
@@ -29,22 +34,29 @@ length limit is based on a survey of the defaults for various JSON parsers.
 parameters are typed.
-  * integer - may be signed or not as specified in the range parameter.
+  * integer - may be signed or not as specified in the range property.
   * string - aligned with JSON strings, these should be UTF-8 encoded and avoid control characters.  
   * enumerated - one of an ordered set of labels which MUST conform to the same limitations as parameter names.  
-  * boolean - may have the values false or true (lower case and unquoted as in JSON)
-  * For integer parameters the range specifies minimum and maximum values, separated by a comma, which MUST be integers themselves.
-  * For string parameters, the range specifies the maximum length of the string as an integer number of characters, not bytes. Again, based on a survey of parsers, try to keep strings shorter than 4095 characters.
+  * boolean - MUST have the value *false* or *true* (lower case and unquoted as in JSON)
+* Range: Further qualifies the valid values for a parameter depending on its type.
+  * For integer parameters, the range specifies minimum and maximum values, separated by a comma, which MUST be integers themselves.
+  * For string parameters, the range specifies the maximum length of the string as an integer number of characters, not bytes. Again, based on a survey of parsers, strings SHOULD be shorter than 4095 characters.
   * For enumerated parameters, the range is a comma separated ordered list of labels that are valid for the parameter. Enumerated parameters are case sensitive and MUST always be quoted, so they are JSON strings. 
-* Default: Default value for parameter, which MUST be consistent with the range limitations. If this is left blank then the parameter is mandatory 
+* Default: Default value for parameter, which MUST be consistent with the range limitations. If this is left blank then the parameter is mandatory (if parameters are defined)  
 
 ## Referencing Parameters
 
-Wherever a parameterised extension is referenced, include any parameters in an accompanying JSON file. If using an extensions directory, the JSON file MUST be named for the extension and included in the directory. For example, the example extension above would have an accompanying file *0000-example-extension.json* which might contain:
+Wherever a parameterised extension is referenced, any parameters MAY be included in an accompanying sidecar JSON file that
+uses the registered name with the filename extension `.json`. If using an 'extensions' directory, the JSON file MUST be 
+included at the top level of the directory and not a subdirectory. Another place that an extension may be referenced is
+`ocfl_layout.json` where the sidecar file should be alongside it in the Storage Root. 
+
+For example, the example extension above would have an accompanying file *0000-example-extension.json* which might contain:
 
     "0000-example-extension.md": {  
```

Comment:
Perhaps not appropriate for this PR, but I'm not sure the parameters json file should have this key "0000-example-extension.md". Wouldn't the contents of `0000-example-extension.md` (in this example) be:

```
{
    "firstExampleParameter": 12,  
    "secondExampleParameter": "Hello",  
    "thirdExampleParameter": "Green"  
}
```

Reply:
I can see potential value in keeping the extension name as the key in this JSON document... particularly in the case where an extension is chaining multiple other extensions together.
..or, for the chaining use case, the chained extensions may also be found in the parameter listing.

Reply:
Unless there's an intention to suggest that you could define multiple extensions in the same file, I don't think there's much value in keeping the key, which makes parsing a little kludgy. I'm not sure it would even work for chaining because you can't infer an ordinal relationship.

Reply:
If the reasons for keeping the key are hypothetical, at best... maybe it should be removed.

Reply:
If this is the only change that is being requested in this PR, perhaps we could allow this PR to move forward and make the change in a separate PR.

Reply:
@rosy1280, @neilsjefferies had expressed the intent to clarify the language around where the `json` file is supposed to be written: https://github.com/OCFL/extensions/pull/13#discussion_r440907134

Reply:
But, yes, the key doesn't necessarily need to be addressed here.

Reply:
Parking this for a later pull request.

Reply:
@pwinckles can you create a ticket for this issue? (or did you already?)

Reply:
Done: https://github.com/OCFL/extensions/issues/22